apps/common/src/python/mediawords/db/__init__.py

```diff
@@ -57,12 +70,34 @@ def connect_to_db() -> DatabaseHandler:
             else:
                 log.info("Out of retries, giving up and exiting...")
 
-                # Don't throw any exceptions because they might be caught by
-                # the try-catch block, and so the caller will just assume that
-                # there was something wrong with the input data and proceed
-                # with processing next item in the job queue (e.g. the next
-                # story). Instead, just quit and wait for someone to restart
-                # the whole app that requires database access.
-                fatal_error(error_message)
+                if retries_config.fatal_error_on_failure():
+                    # Don't throw any exceptions because they might be caught by
+                    # the try-catch block, and so the caller will just assume that
+                    # there was something wrong with the input data and proceed
+                    # with processing next item in the job queue (e.g. the next
+                    # story). Instead, just quit and wait for someone to restart
+                    # the whole app that requires database access.
+                    fatal_error(error_message)
```

Comment:
Well, dunno - the situation here is that by default `connect_to_db()` tries connecting to PostgreSQL for about a minute (60 times while sleeping for 1 second between each retry), so if it didn't manage to connect in that time, it most likely means that something's wrong with PostgreSQL itself or the server that's running PostgreSQL, so simply restarting the app that runs it might not be the remedy.

Maybe the comment is misleading then, do you have any insights?

Reply:
In that case I'd say pass through that advice (e.g. `fatal_error(error_message += '. This indicates a problem with PostgreSQL or the server on which it's running; please [do whatever thing one does to solve such problems]`.') Basically what I think would be useful is guidance in the logging about the steps one should consider taking in case of such an error.

Reply:
Problem is that it's not necessarily "PostgreSQL or the server on which it's running". It could be:

* Network problems
* Docker DNS problems
* Disk problems
* Load problems
* Too many connections to PostgreSQL
* PostgreSQL not accepting new clients due to primary key overflow
* PostgreSQL deliberately shut down due to upgrades / maintenance / migrations / whatever
* PostgreSQL recovering from an unclean shutdown (happens pretty often)
* Database client code is messed up in some way
* ...
* ...
* ...

Therefore, I disagree that we can pinpoint the exact problem here ("something is wrong with the server") and suggest a solution ("restart X") as that would be misleading. Our database doesn't go out just because (thankfully :)). All we can do is state the fact that we're unable to connect and decided to give up.